File: go/sequential_digits.go
package leetcode

import (
	"sort"
)

func sequentialDigits(low int, high int) []int {
	var backtrack func(int, int)
	res := make([]int, 0)
	backtrack = func(num, curr int) {
		if num > 9 {
			return
		}
		curr = curr*10 + num
		if curr >= low {
			if curr > high {
				return
			} else {
				res = append(res, curr)
			}
		}
		backtrack(num+1, curr)
	}

	for i := 1; i < 10; i++ {
		backtrack(i+1, i)
	}
	sort.Ints(res)
	return res
}

File: python/sequential-digits.py
from typing import List


class Solution:
    def sequentialDigits(self, low: int, high: int) -> List[int]:
        n = len(str(high))
        def backtrack(num, curr, res):
            if len(curr) > n:
                return
            if num == 9:
                return
            
            curr += str(num+1)
            curr_num = int(curr)
            if low <= curr_num <= high:
                res.append(curr_num)
            backtrack(num+1, curr, res)
            
            
        res = []
        first = 1
        for i in range(first, 10):
            backtrack(i, str(i), res)
        return list(sorted(res))